pkg/task/restore.go

```diff
@@ -400,8 +400,9 @@ func RunRestoreTiflashReplica(c context.Context, g glue.Glue, cmdName string, cf
 	for _, db := range dbs {
 		tables = append(tables, db.Tables...)
 	}
-	updateCh := utils.StartProgress(
+	updateCh := g.StartProgress(
 		ctx, "RecoverTiflashReplica", int64(len(tables)), !cfg.LogProgress)
+	defer updateCh.Close()
```

Comment:
`defer close` always make progress to 100% even if it fails. Could you close it only if there is no error?